skyportal/handlers/api/public_pages/public_source_page.py

```diff
@@ -140,7 +140,8 @@ async def post(self, source_id):
                         )
                     )
                 data_to_publish["photometry"] = [
-                    photo.to_dict_public() for photo in session.scalars(query).all()
+                    photo.to_dict_public()
+                    for photo in session.scalars(query.distinct()).all()
```

Comment:
very minor comment. But in most places what you call the `query` here is called `stmt` (short for "statement"). That is because `.query()` is also an actual method which makes things a little confusing and prone to errors. It's ok as is, but something to keep in mind next time you open a PR.